openmetadata-service/src/test/java/org/openmetadata/service/resources/EntityResourceTest.java

```diff
@@ -2223,59 +2224,69 @@ protected void deleteTagAndCheckRelationshipsInSearch(TestInfo test)
-        }
-      }
-    }
+    Map<String, Object> sourceAsMap =
+        waitForSyncAndGetFromSearchIndex(
+            entity.getUpdatedAt(), entity.getId(), entityReference.getType());
+    List<String> fqnList =
+        ((List<Map<String, String>>) sourceAsMap.get("tags"))
+            .stream().map(tempMap -> tempMap.get("tagFQN")).toList();
+    assertTrue(fqnList.contains(tagLabel.getTagFQN()));
 
     // delete the tag
     tagResourceTest.deleteEntity(tag.getId(), false, true, ADMIN_AUTH_HEADERS);
 
-    T finalEntity = entity;
-    TestUtils.assertEventually(
-        test.getDisplayName(),
-        () -> {
-          fqnList.clear();
-          SearchResponse afterDeleteResponse;
-          try {
-            afterDeleteResponse =
-                getResponseFormSearch(
-                    indexMapping.getIndexName(Entity.getSearchRepository().getClusterAlias()));
-          } catch (HttpResponseException e) {
-            throw new RuntimeException(e);
-          }
+    Map<String, Object> sourceAsMapAfterDelete =
+        waitForSyncAndGetFromSearchIndex(
+            entity.getUpdatedAt(), entity.getId(), entityReference.getType());
+    List<String> fqnListAfterDelete =
+        ((List<Map<String, String>>) sourceAsMapAfterDelete.get("tags"))
+            .stream().map(tempMap -> tempMap.get("tagFQN")).toList();
+    // check if the added tag if also added in the entity in search
+    assertFalse(fqnListAfterDelete.contains(tagLabel.getTagFQN()));
+  }
 
-          SearchHit[] hitsAfterDelete = afterDeleteResponse.getHits().getHits();
-          for (SearchHit hit : hitsAfterDelete) {
-            Map<String, Object> sourceAsMap = hit.getSourceAsMap();
-            if (sourceAsMap.get("id").toString().equals(finalEntity.getId().toString())) {
-              @SuppressWarnings("unchecked")
-              List<Map<String, String>> listTags =
-                  (List<Map<String, String>>) sourceAsMap.get("tags");
-              listTags.forEach(tempMap -> fqnList.add(tempMap.get("tagFQN")));
-              break;
-            }
-          }
-          // check if the relationships of tag are also deleted in search
-          assertFalse(fqnList.contains(tagLabel.getTagFQN()));
-        });
+  public static Map<String, Object> waitForSyncAndGetFromSearchIndex(
+      Long entityDbLastUpdate, UUID entityId, String entityType) {
+    AtomicReference<Map<String, Object>> responseMap = new AtomicReference<>();
+    Awaitility.await("Wait for Indexes to be updated for the Entity")
+        .pollInterval(Duration.ofMillis(2000L))
+        .atMost(Duration.ofMillis(120 * 1000L)) // 60 iterations for 120 seconds
+        .until(
+            () -> {
+              responseMap.set(getEntityDocumentFromSearch(entityId, entityType));
+              Object esLastUpdateAt = responseMap.get().get("updatedAt");
+              long esLastUpdate = Long.parseLong(esLastUpdateAt.toString());
+              return esLastUpdate >= entityDbLastUpdate;
```

Comment:
Do you think we could have scenarios where 1 doc in he index evaluates the condition to true but not all the doc we'll need to compare are yet indexed?

Reply:
correct @TeddyCr , this is in Progress actually , need to do some more work